test/icastats_test.c.in

```diff
@@ -108,9 +124,11 @@ int main (int argc, char **argv)
 	/* ica_random_number_generate uses ica_drbg if available. Otherwise the
 	 * old prng code is used. */
 	if (check_hw(SHA512_DRNG) == 1 || check_hw(SHA512) == 1)
-		check_icastats(SHA512_DRNG, "DRBG-SHA-512");
+		rc = check_icastats(SHA512_DRNG, "DRBG-SHA-512");
 	else
-		check_icastats(P_RNG, "P_RNG");
+		rc = check_icastats(P_RNG, "P_RNG");
+	if (rc)
+		exit(handle_ica_error(rc, "check_icastats"));
```

Comment:
so here you do an exit? is this intentional?

Reply:
After re-thinking about this I would in fact leave the exits in the main routine. If prereqs for icastats tests are not met, it's ok to exit with an error indication. The original problem that later tests were not executed at all when earlier tests failed, is now fixed. Let's leave it as it is here. 
BTW: I will add a reformatting commit on top of this to fix formatting without any further logic change.